src/ArborX_BruteForce.hpp

```diff
@@ -0,0 +1,124 @@
+ ****************************************************************************/
+
+#ifndef ARBORX_BRUTE_FORCE_HPP
+#define ARBORX_BRUTE_FORCE_HPP
+
+#include <ArborX_AccessTraits.hpp>
+#include <ArborX_Box.hpp>
+#include <ArborX_CrsGraphWrapper.hpp>
+#include <ArborX_DetailsBruteForceImpl.hpp>
+#include <ArborX_DetailsKokkosExtAccessibilityTraits.hpp>
+
+#include <Kokkos_Core.hpp>
+
+namespace ArborX
+{
+
+template <typename MemorySpace>
+class BruteForce
+{
+public:
+  using memory_space = MemorySpace;
+  static_assert(Kokkos::is_memory_space<MemorySpace>::value, "");
+  using size_type = typename MemorySpace::size_type;
+  using bounding_volume_type = Box;
+
+  BruteForce() = default;
+
+  template <typename ExecutionSpace, typename Primitives>
+  BruteForce(ExecutionSpace const &space, Primitives const &primitives);
+
+  KOKKOS_FUNCTION
+  size_type size() const noexcept { return _size; }
+
+  KOKKOS_FUNCTION
+  bool empty() const noexcept { return size() == 0; }
+
+  KOKKOS_FUNCTION
+  bounding_volume_type bounds() const noexcept { return _bounds; }
+
+  template <typename ExecutionSpace, typename Predicates, typename Callback,
+            typename Ignore = int>
+  void query(ExecutionSpace const &space, Predicates const &predicates,
+             Callback const &callback, Ignore = Ignore()) const;
+
+  template <typename ExecutionSpace, typename Predicates,
+            typename CallbackOrView, typename View, typename... Args>
+  std::enable_if_t<Kokkos::is_view<std::decay_t<View>>{}>
+  query(ExecutionSpace const &space, Predicates const &predicates,
+        CallbackOrView &&callback_or_view, View &&view, Args &&... args) const
+  {
+    ArborX::query(*this, space, predicates,
+                  std::forward<CallbackOrView>(callback_or_view),
+                  std::forward<View>(view), std::forward<Args>(args)...);
+  }
+
+private:
+  size_type _size;
+  bounding_volume_type _bounds;
+  Kokkos::View<bounding_volume_type *, memory_space> _bounding_volumes;
+};
+
+template <typename MemorySpace>
+template <typename ExecutionSpace, typename Primitives>
+BruteForce<MemorySpace>::BruteForce(ExecutionSpace const &space,
+                                    Primitives const &primitives)
+    : _size(AccessTraits<Primitives, PrimitivesTag>::size(primitives))
+    , _bounding_volumes(Kokkos::ViewAllocateWithoutInitializing(
+                            "ArborX::BF::bounding_volumes"),
+                        _size)
+{
+  static_assert(
+      KokkosExt::is_accessible_from<MemorySpace, ExecutionSpace>::value, "");
+  Details::check_valid_access_traits(PrimitivesTag{}, primitives);
+  using Access = AccessTraits<Primitives, PrimitivesTag>;
+  static_assert(KokkosExt::is_accessible_from<typename Access::memory_space,
+                                              ExecutionSpace>::value,
+                "Primitives must be accessible from the execution space");
+
+  Kokkos::Profiling::pushRegion("ArborX::BF::BF");
```

Comment:
Not sure how I feel about this abbreviation. `BVH` is much more widely used that `BF`.